### DISCUSSION TITLE
Additional abilities for some creatures

### DISCUSSION THREAD

**Discussion by ghost** [ihhub/fheroes2#7467] [Ideas]
In the original game, I almost never upgrade Paladins, Liches, Mages (very expensive building upgrades, expensive units).
These creatures need to be strengthened somehow. It says https://github.com/ihhub/fheroes2/wiki/Beyond-1.0  that abilities can be changed.
One can suggest adding Crusaders 25% magic resistance, Archmages - double damage against undead, Liches - 20% chance to inflict Disrupting Ray, Power Liches - 30% chance. In addition, the action of the Lich shot can be made similar to Heroes 3 - so that it does not touch the neighboring dead.

**Comment by ihhub** [comment]
This is from https://github.com/ihhub/fheroes2/wiki/Beyond-1.0:

>  Any modifications in existing balance of the game are strictly prohibited.

> we agree that the balance in the game between fractions is noticeable and could be changed by adding new monsters / buildings / spells / events.

**Reply by ghost** [reply]
Hi @ihhub !

Yeah, got it. Then it's probably worth introducing a new building that would make a free upgrade for Lichs and Mages.

**Comment by Branikolog** [comment]
Hi, @Ben80-hero .
We shouldn't change the original behaviour and statistics, as it will confuse old players. Adding new abilities to original creatures is not the best way to improve the game. If you want to add a creature, that has double damage to undead, it should be better a completely new one, or at least upgrade of an existing.
I personally love Arch Magi and Crusaders, since after upgrade they have a speed boost, which is definitely a great advantage! Not to mention that they have increased stats and new special abilities.
Speaking about Liches, Are you sure you ever wanted to change cloud attack logic, if you, lets say, never played Heroes 3?
I believe, that original developers made the cloud attack in such way, to balance necromancers faction. Necromant faction already has tons of advantages, like vampires, cheap, but strong bone dragons and zounds of skeletons (which nearly the strongest tier 1 creatures) raised from killed peasants with necromancy.

**Comment by ghost** [comment]
> Hi, @Ben80-hero . We shouldn't change the original behaviour and statistics, as it will confuse old players. Adding new abilities to original creatures is not the best way to improve the game. If you want to add a creature, that has double damage to undead, it should be better a completely new one, or at least upgrade of an existing. I personally love Arch Magi and Crusaders, since after upgrade they have a speed boost, which is definitely a great advantage! Not to mention that they have increased stats and new special abilities. 

Yes, but how much does it cost! I always find a better use for money :)

> I believe, that original developers made the cloud attack in such way, to balance necromancers faction.

Absolutely not)) Original developers have very gross errors in the balance on more serious issues than Lychee. Improving Liches will have almost no effect on the balance, because before the proposed change, Necromancers simply prefer to play through some other advantages (Bone Dragons, Vampires, Necromancy), almost completely ignoring Liches. After the proposed change, Liches will be used a little more often, that's all (but in general they will remain underdogs).

> Necromant faction already has tons of advantages, like vampires, cheap, but strong bone dragons and zounds of skeletons (which nearly the strongest tier 1 creatures) raised from killed peasants with necromancy.

Without nerfing the necromancy skill, the game will remain highly unbalanced anyway. It is enough to carry out the tests that @ihhub is talking about - on most maps, the AI ​​that controls the Necromancer faction will win everyone. BTW, there is a bug in the original game, due to which the AI ​​almost never raises skeletons after the battle. I fixed this bug, but warned the players that the fix will break the balance.

**Reply by Branikolog** [reply]
Necromants are strong now, as they are. Currently liches often perform one guaranteed attack on the first battle stages. Later their usage is quite limited. Liches are good troops, but really tricky to use in battle. Once you succeeded, the result damage will be quite high! But only in some rare cases. You have to keep in mind the positioning of your own troops each time you attack.
So now you proposing to allow necromancers proceed more attacks with liches in battle. I think it will unbalance this faction. Original developers consider this inconvenience of the only archers in faction, while designing other troops, buildings etc. Allowing them to shoot without hurting other necromant creatures will make them too powerful.
Why do you think we have a magical garden, that can give a player a pack of gems, but there's no object, that gives sulfur?
Why do you think, that there exist some forts, that provide free upgrade to some factions' troops (like Barbarian or Knights, but not for strong Warlocks or Necromants)? 😉 
This IS a kind of balance, original developers planned in the game. It's not an ideal, that's true. But it exists. The balance consists  not only of raw creature stats, but heroes skill distribution, Heroes basic secondary skills, price of buildings and creatures, speed of creatures (affecting hero movement range) and many-many other things. Changing even the insignificant values, based only on personal feelings can potentially change the whole game balance. 
Devoted Heroes 2 players are rather conservative people. Changing even the fewest aspects of the game for no really a valid reason could make them disappointed and sad.
The best approach here is not to change the existing feature, but to add a new one. Let's say, that Liches could have 3rd upgrade, where a new feature, like death cloud attack, would be applied. But I personally love the existing Liches behaviour being dangerous raging entities, that could even kill their allies. :)

**Reply by Mr-Bajs** [reply]
Thats probably the best approach as you say maybe adding a upgrade or a third upgrade to existing faction, that would be only be available with the Ressurection expansion, either you play fheroes2 with OG content or fheroes2 with Ressurection with added content, that i assume is the plan? Though having 2 versions of the game maintained with AI balancing reasons might be to much. 
Would be for once nice to play a game with upgraded peasants that isn't totaly worthless.

**Reply by Branikolog** [reply]
3rd upgrade is a unique feature of Warlock faction and I personally against overusing it too much. There could be dozens of other ways to add something new into the game. 😉

**Reply by ihhub** [reply]
@Mr-Bajs , the expansion will introduce new content without changing any existing statistics of objects. So it won't be 2 versions of the game. It would be a set of maps without and with new content. Knowing this simplifies a lot for existing players.

AI doesn't need to have 2 versions. AI would know how to handle all objects but some will just not appear on old maps.

**Comment by ghost** [comment]
> . So now you proposing to allow necromancers proceed more attacks with liches in battle. I think it will unbalance this faction. 

No way. It is impossible to unbalance a faction that is already completely unbalanced. Especially by Liches, which a rational player will never buy. Only a person "who loves art" will buy them :))

**Reply by Branikolog** [reply]
Hey, @Ben80-hero ! 
In discussion thread it is possible to post individual answers to each post, which produces a conversation tree. :) (write a reply instead of write a comment). I believe this will make conversation with multiple people more structured. ;)
I personally love liches. I use them quite often, when creating a different armies for a special purposes. Like fighting slow enemies. Also Liches are the only troops in necromant faction, that have "very fast" speed, which gives you an advantage of first turn. I'm completely against calling them useless creatures. 😃 
The only disadvantage I feel for them is that they disappear too fast, being not really much armoured. 😄

**Reply by ghost** [reply]
@Branikolog,

I like the fact that you love the game very much and that we can discuss it endlessly ))

**Reply by Mr-Bajs** [reply]
If thats is true or not. I find that assume players always have a min-max playstyle is a very narrow view on the subject. Achive the perfect balance think is pretty hard and probably impossible to achive without sacrificing character and uniqueness of the different factions. Also in aspect of retrogames that they players overall value art and a relaxing time gaming aspects of the game over the competative min-max playstyle. But thats my take on it.

Then if multiplayer competativeness is the issue, when/if multiplayer is working you could implement custom multiplayer rules and handicapsystem to mitigate the biggest issues in terms of balancing.

But they best way would be that fheroes2-engine would implement a mod support feature. To be able to easily toggle on and off whatever rebalancing or content mod the player wants then everybody should be happy.

**Reply by ghost** [reply]
Players who play multiplayer love some clear reference points. And other players too. For example, there are a huge number of mods for WoG/Era, but apart from the authors of these mods, almost no one plays them. I still hope that the fheroes2 team will create an expansion that will not disappoint.

**Comment by ghost** [comment]
@Branikolog ,

It's not at all that the developers had bad ideas. No, the ideas are good, and there is a kind of interesting balance. The problem is specifically in numbers - in game analytics. They missed a lot in a lot of places. Ordinary technical errors due to lack of time.

**Comment by ihhub** [comment]
For all believers that Necromancers are strong and Knights are weakest:

https://github.com/ihhub/fheroes2/assets/19829520/e3f68317-2cec-4fe8-9f7c-0d828beed1a5

**Comment by ghost** [comment]
Hi @ihhub ,

I didn't watch the video until the end, but I believe that the Knight won. However, this video in no way proves the strength or weakness of the factions. Relevant inferences require simulation on many different maps several times each.
